tests/cnf/core/network/metallb/tests/frrk8-tests.go

```diff
@@ -0,0 +1,738 @@
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create config map")
+
+	return masterConfigMap
+}
+
+func verifyExternalAdvertisedRoutes(frrPod *pod.Builder, ipv4NodeAddrList, externalExpectedRoutes []string) {
+	// Get advertised routes from FRR pod, now returned as a map of node IPs to their advertised routes
+	advertisedRoutesMap, err := frr.GetBGPAdvertisedRoutes(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+	Expect(err).ToNot(HaveOccurred(), "Failed to find advertised routes")
+
+	// Iterate through each node in the advertised routes map
+	for nodeIP, actualRoutes := range advertisedRoutesMap {
+		// Split the string of advertised routes into a slice of routes
+		routesSlice := strings.Split(strings.TrimSpace(actualRoutes), "\n")
+
+		// Check that the actual routes for each node contain all the expected routes
+		for _, expectedRoute := range externalExpectedRoutes {
+			matched, err := ContainElement(expectedRoute).Match(routesSlice)
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to match route %s for node %s", expectedRoute, nodeIP))
+
+			Expect(matched).To(BeTrue(), fmt.Sprintf("Expected route %s not found for node %s", expectedRoute, nodeIP))
+		}
+	}
+}
+
+func resetOperatorAndTestNS() {
+	By("Cleaning MetalLb operator namespace")
+
+	metalLbNs, err := namespace.Pull(APIClient, NetConfig.MlbOperatorNamespace)
+	Expect(err).ToNot(HaveOccurred(), "Failed to pull metalLb operator namespace")
+	err = metalLbNs.CleanObjects(
+		tsparams.DefaultTimeout,
+		metallb.GetBGPPeerGVR(),
+		metallb.GetBFDProfileGVR(),
+		metallb.GetBGPPeerGVR(),
+		metallb.GetBGPAdvertisementGVR(),
+		metallb.GetIPAddressPoolGVR(),
+		metallb.GetMetalLbIoGVR(),
+		metallb.GetFrrConfigurationGVR())
+	Expect(err).ToNot(HaveOccurred(), "Failed to remove object's from operator namespace")
+
+	By("Cleaning test namespace")
+
+	err = namespace.NewBuilder(APIClient, tsparams.TestNamespaceName).CleanObjects(
+		tsparams.DefaultTimeout,
+		pod.GetGVR(),
+		service.GetServiceGVR(),
+		configmap.GetGVR(),
+		nad.GetGVR())
+	Expect(err).ToNot(HaveOccurred(), "Failed to clean test namespace")
+}
+
+func buildRoutesMapWithSpecificRoutes(podList []*pod.Builder, nextHopList []string) map[string]string {
+	Expect(len(podList)).ToNot(BeZero(), "Pod list is empty")
+	Expect(len(nextHopList)).ToNot(BeZero(), "Nexthop IP addresses list is empty")
+	Expect(len(nextHopList)).To(BeNumerically(">=", len(podList)),
+		fmt.Sprintf("Number of speaker IP addresses[%d] is less than the number of pods[%d]",
+			len(nextHopList), len(podList)))
+
+	routesMap := make(map[string]string)
+
+	for _, frrPod := range podList {
+		if frrPod.Definition.Spec.NodeName == workerNodeList[0].Definition.Name {
+			routesMap[frrPod.Definition.Spec.NodeName] = nextHopList[1]
+		} else {
+			routesMap[frrPod.Definition.Spec.NodeName] = nextHopList[0]
+		}
+	}
+
+	return routesMap
+}
+
+func verifyFrrWebHookServerRunning() {
```

Comment:
instead of this function. try to use something like:
```
			By("Waiting until the new frr-k8s-webhook-server deployment is in Ready state.")
			frrk8sWebhookDeployment, err := deployment.Pull(
				APIClient, "frr-k8s-webhook-server", NetConfig.MlbOperatorNamespace)
			Expect(frrk8sWebhookDeployment.IsReady(30 * time.Second)).To(BeTrue(), "frr-k8s-webhook-server deployment is not ready")
```